### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by edrickhoo**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from "react";
+import { fetchBooksApiSearch, bookType } from "./api/google-books-api";
+import "./App.css";
+import Header from "./components/Header/Header";
+
+import Books from "./container/Books-container/Books";
+
+function App() {
+  const [booksData, setBooksData] = useState<bookType[] | null>(null);
```
Comment (ABaulderstone) [inline src/App.tsx]:
```ts
const [booksData, setBooksData] = useState<bookType[] >([]);
```
Empty array is still technically an array of `bookType`, which means no need for null check just a check on length of array

File: src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  const [error, setError] = useState<string | null>(null);
```
Comment (ABaulderstone) [inline src/App.tsx]: Similarly 
```ts
const [error, setError] = useState<string>("");
```
Empty string is still a falsy value. No need for union type

File: src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    e: React.SyntheticEvent
+  ) => {
+    try {
+      e.preventDefault();
+      setError(null);
+      setBooksData(null);
+      const response = await fetchBooksApiSearch(searchInput);
+
+      setBooksData(response);
+    } catch (err: any) {
```
Comment (ABaulderstone) [inline src/App.tsx]: err: Error should be fine here

File: src/api/google-books-api.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  author: [];
+  description: string;
+  title: string;
+  publishedDate: string;
+  languages: string;
+  rating: string;
+  ratingCount: string;
+  shortDescription: string;
+  pageCount: string;
+};
```
Comment (ABaulderstone) [inline src/api/google-books-api.tsx]: Generally prefer interfaces to types. Naming convention is similar to classes
```ts
export interface Book = {
  img: string;
  author: [];
  description: string;
  title: string;
  publishedDate: string;
  languages: string;
  rating: string;
  ratingCount: string;
  shortDescription: string;
  pageCount: string;
};
```

File: src/components/Book/Book.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+const Book = ({ bookData, setAndToggleMoreInfo }: Props) => {
+  const { img, author, title, shortDescription } = bookData;
+
+  return (
+    <div className={styles.Book}>
+      <h4>{title}</h4>
+      <h5>
+        {Array.isArray(author) && author?.length > 1
+          ? `Authors: ${author.join(", ")}`
+          : `Author: ${author}`}
```
Comment (ABaulderstone) [inline src/components/Book/Book.tsx]: You can do this in your organiseData function

File: src/container/Books-container/Books.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+          <img
+            className={styles.Search_Form__Submit__Icon}
+            src={searchIcon}
+            alt="Search Icon"
+          />
+        </button>
+      </form>
+      <div className={styles.Books}>
+        {booksData && booksData.length > 0 ? (
+          booksData.map((book: any, idx: number) => {
```
Comment (ABaulderstone) [inline src/container/Books-container/Books.tsx]: Can avoid the null check with the previous type change. 
`book: bookType` or better yet `book: Book`

File: src/container/Books-container/Books.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        ) : (
+          <div className={styles.Search_Error}>
+            {error ? error : "Loading..."}
+          </div>
+        )}
+      </div>
+
+      {moreInfo ? (
+        <MoreInfo toggleMoreInfo={toggleMoreInfo} moreInfoData={moreInfoData} />
+      ) : null}
```
Comment (ABaulderstone) [inline src/container/Books-container/Books.tsx]:
```ts
 {moreInfo && moreInfoData && (
        <MoreInfo toggleMoreInfo={toggleMoreInfo} moreInfoData={moreInfoData} />
      )}
 ```
 You need the null check for moreInfoData to satisfy that the type you're passing into <MoreInfo> isn't null. No need for a ternary if you're just rendering null as the alternative

**Review by ABaulderstone — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (edrickhoo) [reply src/App.tsx]: Fixed

File: src/App.tsx (hunk 2 above)
Comment (edrickhoo) [reply src/App.tsx]: Fixed

File: src/api/google-books-api.tsx (hunk 4 above)
Comment (edrickhoo) [reply src/api/google-books-api.tsx]: Fixed

File: src/container/Books-container/Books.tsx (hunk 6 above)
Comment (edrickhoo) [reply src/container/Books-container/Books.tsx]: Fixed

File: src/container/Books-container/Books.tsx (hunk 7 above)
Comment (edrickhoo) [reply src/container/Books-container/Books.tsx]: Fixed

File: src/components/Book/Book.tsx (hunk 5 above)
Comment (edrickhoo) [reply src/components/Book/Book.tsx]: Fixed

File: src/App.tsx (hunk 3 above)
Comment (edrickhoo) [reply src/App.tsx:44]: Tried to change still get an Error

image

**Pull request merged**